tests/main/failover/task.yaml

```diff
@@ -87,10 +90,17 @@ execute: |
         # repack new target snap
         snap pack "$BUILD_DIR/unpack" && mv ${TARGET_SNAP_NAME}_*.snap failing.snap
 
+        if check_journalctl_log "Waiting for system reboot"; then
+            echo "Already waiting for system reboot, exiting..."
+            exit 1
+        fi
+
         # install new target snap
         snap install --dangerous failing.snap
 
-        while ! journalctl -b -u snapd|grep -q "Waiting for system reboot" ; do sleep 1 ; done
```

Comment:
Wonder why `journalctl -b` didn't work as expected. In theory those are the logs from current boot only.